src/extension/chatSessions/vscode-node/copilotCLIChatSessionsContribution.ts

```diff
@@ -338,26 +316,29 @@ export class CopilotCLIChatSessionParticipant {
 			return await this.handlePushConfirmationData(request, context, stream, token);
 		}
 
+		const defaultModel = await this.copilotCLIModels.getDefaultModel();
+		const { resource } = chatSessionContext.chatSessionItem;
+		const id = SessionIdForCLI.parse(resource);
+		const preferredModel = _sessionModel.get(id);
+		// For existing sessions we cannot fall back, as the model would info would be updated in _sessionModel
+		const modelId = this.copilotCLIModels.toModelProvider(preferredModel?.id || defaultModel.id);
+		const { prompt, attachments } = await this.promptResolver.resolvePrompt(request, token);
+
 		if (chatSessionContext.isUntitled) {
-			const untitledCopilotcliSessionId = SessionIdForCLI.parse(chatSessionContext.chatSessionItem.resource);
-			const workingDirectory = await this.worktreeManager.createWorktreeIfNeeded(untitledCopilotcliSessionId, stream);
+			const session = await this.sessionService.createSession(prompt, modelId, token);
+			const workingDirectory = await this.worktreeManager.createWorktreeIfNeeded(session.sessionId, stream);
```

Comment:
@rebornix Something doesn't look right here. 
If previously we were using `untitledCopilotCliSessionId`, surely we cannot use a new session (valid) id here?

We can sync on this next week, I'm happy to merge this for now.